opencog/rule-engine/backwardchainer/BackwardChainer.cc

```diff
@@ -36,20 +36,12 @@
 
 using namespace opencog;
 
-BackwardChainer::BackwardChainer(AtomSpace* as, const std::vector<Rule>& rs)
-	: _as(as), _rules_set(rs)
```

Comment:
Umm... why did you remove the storing the vector of rule objects?  Now _rules_set is empty and no rule is given to the BackwardChainer, break the unit test.